### PR TITLE
Inline captured variables

### PR DESCRIPTION
**Pull request by francescoo22**
- I think we can also use the `scopeDepth` for `returnLabelIndex` instead of `programCtx.newtReturnLabelIndex()`
- Let me know if you have some better idea for avoiding clashes in `addScopedName` and `getScopeDepth`

### CODE REVIEW

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConversionContext.kt [hunk 1]
```diff
@@ -27,6 +27,7 @@ interface ProgramConversionContext {
     fun getField(field: FirPropertySymbol): FieldEmbedding?
     fun newAnonName(): AnonymousName
     fun newWhileIndex(): Int
+    fun newtReturnLabelIndex(): Int
```
Comment (jesyspa) [inline plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConversionContext.kt:30]: typo: newt

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/InlineMethodConverter.kt [hunk 2]
```diff
@@ -20,25 +21,26 @@ sealed interface SubstitutionItem {
...
 data class SubstitutionName(override val name: MangledName) : SubstitutionItem
 
-data class SubstitutionLambda(val body: FirBlock, val args: List<Name>) : SubstitutionItem {
+data class SubstitutionLambda(val body: FirBlock, val args: List<Name>, val scopedNames: Map<Name, Int>) : SubstitutionItem {
     override val name = null
     override fun lambdaBody(): FirBlock = body
     override fun lambdaArgs(): List<Name> = args
 }
 
 class InlineNameMangler(
```
Comment (jesyspa) [inline plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/InlineMethodConverter.kt:30]: Would it be possible to switch to having a single mangler?

File: plugins/formal-verification/testData/diagnostics/no_contracts/inlining_captured.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
+inline fun <!VIPER_TEXT!>invoke<!>(f: (Int) -> Int): Int {
```
Comment (jesyspa) [inline plugins/formal-verification/testData/diagnostics/no_contracts/inlining_captured.kt:1]: If you don't care about the generated code, you could use `NeverConvert` to drop this from the diagnostics output.

**Review by jesyspa — APPROVED** [review]
Looks good, just some minor stuff

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/InlineMethodConverter.kt (hunk 2 above)
Comment (francescoo22) [reply plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/InlineMethodConverter.kt:30]: I think that we can do it

**Pull request merged**